src/main/java/edu/cornell/kfs/krad/util/MaintenanceLockUtils.java

```diff
@@ -0,0 +1,52 @@
+package edu.cornell.kfs.krad.util;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.sys.KFSConstants;
+
+public final class MaintenanceLockUtils {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private static final int BULK_LOCK_QUERY_LIMIT = 500;
+
+    private MaintenanceLockUtils() {
+        throw new UnsupportedOperationException("do not call");
+    }
+
+    public static String doChunkedSearchForAnyLockingDocumentNumber(List<String> lockingRepresentations,
+            String documentNumber, BiFunction<List<String>, String, String> actualQueryOperation) {
+        if (CollectionUtils.isEmpty(lockingRepresentations)) {
+            LOG.debug("doChunkedSearchForAnyLockingDocumentNumber, No locking reps specified, skipping search");
+            return null;
+        }
+        LOG.debug("doChunkedSearchForAnyLockingDocumentNumber, Searching with {} lock reps{}",
+                lockingRepresentations::size,
+                () -> StringUtils.isNotBlank(documentNumber)
+                        ? " and excluding results owned by document " + documentNumber
+                        : KFSConstants.EMPTY_STRING);
+        
+        String lockingDocumentNumber = null;
+        int startIndex = 0;
+        
+        do {
+            int endIndex = Math.min(startIndex + BULK_LOCK_QUERY_LIMIT, lockingRepresentations.size());
+            List<String> locksSubList = lockingRepresentations.subList(startIndex, endIndex);
+            lockingDocumentNumber = actualQueryOperation.apply(locksSubList, documentNumber);
+            startIndex += BULK_LOCK_QUERY_LIMIT;
```

Comment:
based on this code is structured, I don't think it matters, but wouldn't it mkae more sense to add endIndex to startIndex?  The next iteration starts with where the current iteration started plus where it ended.

Reply:
It shouldn't make much of a difference, but I can adjust this to set the start index to the end index.